cloud/amazon/execute_lambda.py

```diff
@@ -0,0 +1,273 @@
+        the task will not return any results. To wait for the Lambda function
+        to complete, set C(await=true) and the result will be available in the
+        I(output) key.
+    required: false
+    default: true
+  dry_run:
+    description:
+      - Do not *actually* invoke the function. A C(DryRun) call will check that
+        the caller has permissions to call the function, especially for
+        checking cross-account permissions.
+    required: false
+    default: False
+  version_qualifier:
+    description:
+      - Which version/alias of the function to run. This defaults to the
+        C(LATEST) revision, but can be set to any existing version or alias.
+        See https;//docs.aws.amazon.com/lambda/latest/dg/versioning-aliases.html
+        for details.
+    required: false
+    default: LATEST
+  payload:
+    description:
+      - A dictionary in any form to be provided as input to the Lambda function.
+    required: false
+    default: {}
+'''
+
+EXAMPLES = '''
+- execute_lambda:
+    name: test-function
+    # the payload is automatically serialized and sent to the function
+    payload:
+      foo: bar
+      value: 8
+  register: response
+
+# Test that you have sufficient permissions to execute a Lambda function in
+# another account
+- execute_lambda:
+    function_arn: arn:aws:lambda:us-east-1:123456789012:function/some-function
+    dry_run: true
+
+- execute_lambda:
+    name: test-function
+    payload:
+      foo: bar
+      value: 8
+    await: true
+    tail_log: true
+  register: response
+  # the response will have a `logs` key that will contain a log (up to 4KB) of the function execution in Lambda.
+
+- execute_lambda: name=test-function version_qualifier=PRODUCTION
+'''
+
+RETURN = '''
+output:
+    description: Function output if await=true and the function returns a value
+    returned: success
+    type: dict
+    sample: "{ 'output': 'something' }"
+logs:
+    description: The last 4KB of the function logs. Only provided if I(tail_log) is true
+    type: string
+status:
+    description: C(StatusCode) of API call exit (200 for synchronous invokes, 202 for async)
+    type: int
+    sample: 200
+'''
+
+import base64
+import json
+import traceback
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name                 = dict(),
+        function_arn         = dict(),
+        await                = dict(choices=BOOLEANS, default=True, type='bool'),
+        tail_log             = dict(choices=BOOLEANS, default=False, type='bool'),
+        dry_run              = dict(choices=BOOLEANS, default=False, type='bool'),
+        version_qualifier    = dict(),
+        payload              = dict(default={}, type='dict'),
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
```

Comment:
The module sets ```supports_check_mode=True```, but there is no logic to handle check mode. Shouldn't check mode be used to enable a dry run?